src/axom/primal/geometry/Polygon.hpp

```diff
@@ -22,63 +22,184 @@ namespace axom
 {
 namespace primal
 {
+namespace
+{
+static constexpr int DEFAULT_MAX_NUM_VERTICES = 20;
+
+/*!
+ * \brief A helper class for the Polygon class
+ *
+ * This class helps define a StackArray with a current size and additional
+ * functions to mirror axom::Array.
+ */
+template <typename T, int N>
+struct StaticArray : public StackArray<T, N>
+{
+  AXOM_HOST_DEVICE int size() const { return m_size; }
+  AXOM_HOST_DEVICE void push_back(const T& obj)
+  {
+    SLIC_ASSERT(m_size + 1 <= N);
+    StackArray<T, N>::m_data[m_size++] = obj;
+  }
```

Comment:
Note: `SLIC_ASSERT` is only enabled in debug configs.

Should there be a runtime check when `push_back` is called an the array is full?
At the very least, we should not write to `m_data[N]` and beyond.

My concern is that this will silently overwrite memory is should not write to in release configuration.

Reply:
I'll leave this as a debug-only check for now.
This may become a runtime check in a follow-up PR for polygon clipping.

I changed this check to an `assert()` for now (remove `slic` dependency from `core` if/when `StaticArray` is moved to `core`)

Reply:
Thanks @bmhan12 

It might be better to have something like this:
```cpp
    assert(m_size + 1 <= N);
    if(m_size <  N-1)
    { 
      StackArray<T, N>::m_data[m_size++] = obj; 
    }
```

or to throw an error / asm trap if writing out-of-bounds, since this would help localize the bug.

I wouldn't expect it to add too much overhead considering the cost/benefit.

Typically, for arrays, the subscript operator doesn't have checks, but I'd expect something like `push_back()` to have a check.